tools/index.ts

```diff
@@ -186,4 +199,65 @@ export default class LexicalModelCompiler {
   logError(s) {
     console.error(require('chalk').red(s));
   };
-};
\ No newline at end of file
+};
+
+/**
+ * Returns a data structure suitable for use by the trie wordlist model.
+ *
+ * Format specification:
+ *
+ *  - the file is a UTF-8 encoded text file
+ *  - new lines are either LF or CRLF
+ *  - the file either consists of a comment or an entry
+ *  - comment lines MUST start with the '#' character
+ *  - entries are one to three columns, separated by the (horizontal) tab
+ *    character
+ *  - column 1 (REQUIRED): the wordform: can have any character except tab, CR,
+ *    LF. surrounding SPACE characters are trimmed.
+ *  - column 2 (optional): the count: a non-negative integer specifying how many
+ *    times this entry has appeared in the corpus. Blank means 'indeterminate'
+ *  - column 3 (optional): comment: an informative comment, ignored by the tool.
+ *
+ * @param sourceFiles an array of the CONTENTS of source files
+ * @return a data structure that will be used internally by the trie wordlist
+ *         implemention. Currently this is an array of [wordlist, count] pairs.
+ */
+function createTrieDataStructure(sourceFiles: string[]) {
```

Comment:
I'd like to see us putting model-specific compilation code in their own .ts files, since index.ts could get rather cluttered if/when a lot of models get added.  We can always link 'em in via `///<reference path="___" />`.

Reply:
+1